_posts/2017-12-27-cheap-docker-swarm-cluster-aws-terraform.md

```diff
@@ -0,0 +1,98 @@
+
+To summarize:
+
+| Description | Cost |
+| ----------- | ---- |
+| Monthly Amortized Hardware Costs | $20.83 |
+| Monthly Electricity Usage        | $21.60 |
+| Total                            | $42.43 |
+
+## Cloud Hosting
+
+With [AWS](https://aws.amazon.com/), there are a large number of server types to choose from.  A comparable EC2 type to our self-hosted solution would be the t2.medium instance.  [Here’s a great site that compares the various options and prices.](https://www.ec2instances.info/)  If we use on-demand pricing, we will get higher availability, but we are paying full price.  We have the option to go with spot pricing and we are going to sacrifice availability for some savings.  Here’s the comparison for the t2.medium, with current 2017 pricing.
+
+
+| Lifecycle | Total Hourly Cost | Total Monthly Cost |
+| --------- | ----------------- | ------------ |
+| On-Demand | $0.0464 | $33.41 |
+| Spot | $0.0185 | $13.32 |
+
+Even with full availability, we are still getting cheaper prices than our self-build, but a significant savings when we look at spot pricing.
+
+## Setting it all up
+
+Another major factor we need to discuss is how long it takes to set this stuff up.  We’ve done a simple comparison which makes a compelling case for the cloud.  But where the cloud really shines is in the time savings setting the cluster up.
+
+It look me about 2 hours to build each of my self-hosted boxes, that’s after sourcing all of the parts.  Let’s say a total of 6 hours, assuming everything goes well.
+
+How long does it take on the cloud?  With a little help from [Terraform](https://www.terraform.io/), we can have that same setup in about 5 minutes.
+
+So let’s factor that in as well.  How you value your time is certainly up to you, but to try to keep costs low, let’s assume a $10 minimum hourly wage for the setup time.  That’s an extra $60 of upfront setup time.  Over the course of the lifetime of the servers however, that’s not really that big of a deal, but it does add up.  We could also talk about the cost of maintenance if something goes wrong.  If you’re self hosting, you’re on the hook to fix stuff, unless you need an expensive paper weight.   In the Cloud, in a matter of minutes, you can swap out a bad instance with a fresh new one.
+
+## Show me the code
+
+If you would like to spin something up quickly using spot instances, go ahead and checkout the code.
+
+[https://github.com/rrx/aws-docker-swarm](https://github.com/rrx/aws-docker-swarm)
+
+The code is a fork of some well written terraform by another developer.  It was so well done, that I didn’t think it was worth it to reinvent the wheel.  I just modified it to use spot instances.
+
+Here are some quick instructions to get you up and running.
```

Comment:
This should have been done earlier, because by golly when you made a reference to it I thought you were pulling things out of your rear end as "proof."